### PR TITLE
Sends notifications for public posts with "uid=0"

### PR DESCRIPTION
**Pull request by annando**
See https://github.com/friendica/friendica/issues/4117

### CODE REVIEW

**Comment by annando** [comment]
I added a change so that dropping of our items is possible as well.

File: include/enotify.php [hunk 1]
```diff
@@ -106,9 +106,8 @@ function notification($params)
 	}
 
 	if ($params['type'] == NOTIFY_COMMENT) {
-		$p = q("SELECT `ignored` FROM `thread` WHERE `iid` = %d AND `uid` = %d LIMIT 1",
-			intval($parent_id),
-			intval($params['uid'])
+		$p = q("SELECT `ignored` FROM `thread` WHERE `iid` = %d LIMIT 1",
```
Comment (MrPetovan) [inline include/enotify.php]: No intention to convert this to `dba:select` while you're at it?

Comment (MrPetovan) [inline include/enotify.php]: No intention to convert this to `dba:select` while you're at it?

File: include/enotify.php [hunk 2]
```diff
@@ -134,9 +133,8 @@ function notification($params)
 		$p = null;
 
 		if ($params['otype'] === 'item' && $parent_id) {
-			$p = q("SELECT * FROM `item` WHERE `id` = %d AND `uid` = %d LIMIT 1",
-				intval($parent_id),
-				intval($params['uid'])
+			$p = q("SELECT * FROM `item` WHERE `id` = %d LIMIT 1",
```
Comment (MrPetovan) [inline include/enotify.php]: No intention to convert this to `dba:select` while you're at it?

File: include/enotify.php [hunk 3]
```diff
@@ -735,9 +749,8 @@ function check_item_notification($itemid, $uid, $defaulttype = "") {
 
 	if ($item[0]["parent-uri"] === $item[0]["uri"]) {
 		// Send a notification for every new post?
-		$r = q("SELECT `notify_new_posts` FROM `contact` WHERE `id` = %d AND `uid` = %d AND `notify_new_posts` LIMIT 1",
-			intval($item[0]['contact-id']),
-			intval($uid)
+		$r = q("SELECT `notify_new_posts` FROM `contact` WHERE `id` = %d AND `notify_new_posts` LIMIT 1",
```
Comment (MrPetovan) [inline include/enotify.php]: No intention to convert this to `dba:select` while you're at it?

File: include/enotify.php (hunk 1 above)
Comment (annando) [reply include/enotify.php]: Currently I'm more in the mood to enhance this public item stuff. There is much work to do.

Comment (MrPetovan) [reply include/enotify.php]: Gotcha.

**Comment by MrPetovan** [comment]
Are you done with this PR?

**Comment by annando** [comment]
Now.

File: src/Database/DBStructure.php [hunk 4]
```diff
@@ -1601,7 +1601,7 @@ public static function definition() {
 						"uid_created" => array("uid","created"),
 						"uid_commented" => array("uid","commented"),
 						"uid_wall_created" => array("uid","wall","created"),
-						"private_wall_received" => array("private","wall","received"),
+						"private_wall_commented" => array("private","wall","commented"),
```
Comment (rabuzarus) [inline src/Database/DBStructure.php]: Please add a description in friendicas db documentation

Comment (annando) [reply src/Database/DBStructure.php]: We don't describe indexes, don't we?

Comment (rabuzarus) [reply src/Database/DBStructure.php]: Ohh sry. My fault

**Pull request merged**